docs/DeveloperGuide.md

```diff
@@ -403,11 +402,139 @@ Use case ends.
+
+Use case ends.
+
+**Extensions**
+
+* 2a. Invalid index number
+    * 2a1. NetBook displays an error message stating that the given index is invalid
+    * 2a2. User re-enters delete command with valid index
+    * 2a3. Steps 2a1 and 2a2 are repeated until the information entered is valid
+    * **Use case resumes from Step 3**
+* 2b. Invalid Data
+    * 2b1. NetBook displays an error message informing the user of the correct format
+    * 2b2. User re-enters data in the correct format
+    * 2b3. Steps 2b1 and 2b2 are repeated until the data entered is valid
+    * **Use case resumes from Step 3**
+
+***
+**Use case:** UCO8 - Sort Contacts
+**Actor:** User
+
+**Main Success Scenario (MSS)**
+1. User requests to sort contacts by a specific sorting preference
+2. NetBook validates the input from the user
+3. NetBook updates the GUI so that the contacts are arranged according to the sorting preference
+
+Use case ends.
+
+**Extensions**
+
+* 2a. Invalid sorting preference
+    * 2a1. NetBook displays an error message stating that the given sorting preference is invalid
+    * 2a2. User re-enters delete command with a valid sorting preference
+    * 2a3. Steps 2a1 and 2a2 are repeated until the sorting preference entered is valid
+    * **Use case resumes from Step 3**
+
+***
+**Use case:** UCO9 - Save sorting preference
+**Actor:** User
+
+**Main Success Scenario (MSS)**
+1. User requests to save a specific sorting preference
+2. NetBook validates the input from the user
+3. NetBook updates the saved sorting preference
+
+Use case ends.
+
+**Extensions**
+
+* 2a. Invalid sorting preference
+    * 2a1. NetBook displays an error message stating that the given sorting preference is invalid
+    * 2a2. User re-enters delete command with a valid sorting preference
+    * 2a3. Steps 2a1 and 2a2 are repeated until the sorting preference entered is valid
+    * **Use case resumes from Step 3**
+
+***
+**Use case:** UC10 - Add a remark to a contact
+**Actor:** User
+
+**Main Success Scenario (MSS)**
+1. User requests to add a remark to a contact
+2. NetBook validates the input from the user
+3. NetBook adds the given remark to the contact
+4. NetBook displays the remark in the GUI
+
+Use case ends.
+
+**Extensions**
+
+* 2a. Invalid index number
+    * 2a1. NetBook displays an error message stating that the given index is invalid
+    * 2a2. User re-enters delete command with valid index
+    * 2a3. Steps 2a1 and 2a2 are repeated until the information entered is valid
+    * **Use case resumes from Step 3**
+
+***
+**Use case:** UC11 - Create a reminder
+**Actor:** User
+
+**Main Success Scenario (MSS)**
+1. User requests to create a reminder about a contact
+2. NetBook validates the input from the user
+3. NetBook creates the given reminder
+4. NetBook displays the remark in the GUI
+
+Use case ends.
+
+**Extensions**
+
+* 2a. Invalid index number
+    * 2a1. NetBook displays an error message stating that the given index is invalid
```

Comment:
I think we can just say `displays an error message` because we are not suppose to give too specific implementation details

Reply:
Ok, changed all instances